### PR TITLE
feat/enterpriseportal: UpdateCodyGatewayAccess RPC definition

### PR DESCRIPTION
**Pull request by bobheadxi**
Proposed RPC definition for updating Cody Gateway access, which is currently part of the `updateProductSubscription` GraphQL mutation, via Enterprise Portal. The RPC follows patterns suggested for "update" RPCs in https://google.aip.dev/134.

As an example, to use `UpdateCodyGatewayAccess` to update a subscription's code completions limit:

```ts
UpdateCodyGatewayAccess({
  access: {
    subscription_id: "es_...",
    code_completions: { limit: 420 },
  },
  update_mask: {
    paths: ["code_completions.limit"],
  },
})  
```

Closes CORE-154. Implementation is tracked in a follow-up (CORE-159) - for now this will just return "unimplemented".

I've also removed the "draft state" docstrings.

## Test plan

CI

### CODE REVIEW

File: lib/enterpriseportal/codyaccess/v1/codyaccess.proto [hunk 1]
```diff
@@ -22,6 +21,14 @@ service CodyAccessService {
   rpc ListCodyGatewayAccesses(ListCodyGatewayAccessesRequest) returns (ListCodyGatewayAccessesResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
+
+  // UpdateEnterpriseSubscription updates the Cody Gateway access granted to an
+  // Enterprise subscription.
+  //
+  // Only properties specified by the update_mask are applied.
```
Comment (unknwon) [inline lib/enterpriseportal/codyaccess/v1/codyaccess.proto:28]: After last time we discussed in Slack based on the following

image

My understanding of "AIP" principle for update is:

1. When the `update_mask` is absent, non-empty fields are updated, e.g.
	CleanShot 2024-06-18 at 09 36 01@2x
1. When specified, fields are updated by the list of "path"s, e.g.
	CleanShot 2024-06-18 at 09 36 45@2x
1. When `*` is used as "path", force update all fields, e.g.
	CleanShot 2024-06-18 at 09 37 12@2x

Following that, in the RPC docstring, we'd better to say which exact paths are updatable, e.g. (you're doing it already in the PR)

CleanShot 2024-06-18 at 09 38 03@2x


All of which is to say "Only properties specified by the update_mask are applied." should be removed 😂

**Review by unknwon — APPROVED** [review]

File: lib/enterpriseportal/codyaccess/v1/codyaccess.proto (hunk 1 above)
Comment (bobheadxi) [reply lib/enterpriseportal/codyaccess/v1/codyaccess.proto:28]: Makes sense! I copied this from the existing `UpdateEnterpriseSubscription` 😛 I'll update both

Comment (unknwon) [reply lib/enterpriseportal/codyaccess/v1/codyaccess.proto:28]: The one for `UpdateEnterpriseSubscription` is taken cared in my PR, maybe just update this one so either PR merges would not create conflict 😁

Comment (bobheadxi) [reply lib/enterpriseportal/codyaccess/v1/codyaccess.proto:28]: ah, okay haha

**Pull request merged**